File: Data Structures & Algorithms/number-of-sub-arrays-of-size-k-and-average-greater-than-or-equal-to-threshold/submission-0.java
public class Solution {

    public int numOfSubarrays(int[] arr, int k, int threshold) {
        int left = 0;
        int right = 0;
        int res = 0;
        int currSum = 0;

        while (right < arr.length) {
            if (right - left < k) {
                currSum += arr[right];
                right++;

                if (right - left == k) {
                    int avr = currSum / k;

                    if (avr >= threshold) {
                        res++;
                    }
                }
            } else {
                currSum -= arr[left];
                left++;
            }
        }

        return res;
    }
}

File: Data Structures & Algorithms/eating-bananas/submission-0.kt
class Solution {
    fun minEatingSpeed(piles: IntArray, h: Int): Int {
        var right = piles.max()
        var res = Int.MAX_VALUE
        var left = 1

        while (left <= right) {
            val mid = (left + right) / 2
            if (check(piles, h, mid)) {
                res = mid
                right = mid - 1
            } else {
                left = mid + 1
            }
        }

        return res
    }

    private fun check(piles: IntArray, h: Int, curr: Int): Boolean {
        var hoursSpent = 0

        for (pile in piles) {
            hoursSpent += (pile / curr + if (pile % curr == 0) 0 else 1)
        }

        return hoursSpent <= h
    }
}